source/common/filesystem/BUILD

```diff
@@ -8,6 +8,38 @@ load(
 
 envoy_package()
 
+envoy_cc_library(
+    name = "directory_iterator_lib",
+    srcs = select({
```

Comment:
Could we have two distinct libraries here?

Reply:
Are you suggesting we have separate `directory_iterator_<x>_lib` libraries for Windows / POSIX? If we did that, any users of this library would have to use `select` to pick the right one. We did it this way to follow the pattern used for the `watcher_lib`

Reply:
Yeah, that doesn't sound great either. What about adding some Bazel magic? We could have a `envoy_cc_platform_library` macro that takes a `name` parameter, and then switches to the Windows/Linux versions depending. I.e. bury the `select` in there.